ui/src/containers/VolumePage.js

```diff
@@ -0,0 +1,750 @@
+            storage={pV?.spec?.capacity?.storage ?? intl.translate('unknown')}
+            status={volumeStatus ?? intl.translate('unknown')}
+            storageClassName={volume?.spec?.storageClassName}
+            creationTimestamp={volume?.metadata?.creationTimestamp}
+            volumeType={
+              volume?.spec?.rawBlockDevice
+                ? RAW_BLOCK_DEVICE
+                : SPARSE_LOOP_DEVICE
+            }
+            usedPodName={UsedPod?.name}
+            devicePath={
+              volume?.spec?.rawBlockDevice?.devicePath ??
+              intl.translate('not_applicable')
+            }
+            volumeUsagePercentage={currentVolume?.usage}
+            volumeUsageBytes={currentVolume?.usageRawData ?? 0}
+            storageCapacity={
+              volumeListData?.find((vol) => vol.name === currentVolumeName)
+                .storageCapacity
+            }
+            health={
+              volumeListData?.find((vol) => vol.name === currentVolumeName)
+                .health
+            }
+          ></VolumeDetailCard>
+          <ActiveAlertsCard
+            alertlist={alertlist}
+            PVCName={PVCName}
+          ></ActiveAlertsCard>
+          <PerformanceGraphCard
+            volumeUsed={
+              volumeUsed?.length
+                ? addMissingDataPoint(
+                    jointDataPointBaseonTimeSeries(volumeUsed),
+                    queryStartingTime,
+                    7,
+                    1,
+                  )
+                : null
+            }
+            volumeStorageCapacity={allSizeUnitsToBytes(
+              pV?.spec?.capacity?.storage,
+            )}
+            volumeLatency={
+              volumeLatency
+                ? addMissingDataPoint(
+                    jointDataPointBaseonTimeSeries(volumeLatency),
+                    queryStartingTime,
+                    7,
+                    1,
+                  )
+                : null
+            }
+            volumeThroughputWrite={
+              volumeThroughputWrite
+                ? addMissingDataPoint(
+                    jointDataPointBaseonTimeSeries(volumeThroughputWrite),
+                    queryStartingTime,
+                    7,
+                    1,
+                  )
+                : null
+            }
+            volumeThroughputRead={
+              volumeThroughputRead
+                ? addMissingDataPoint(
+                    jointDataPointBaseonTimeSeries(volumeThroughputRead),
+                    queryStartingTime,
+                    7,
+                    1,
+                  )
+                : null
+            }
+            volumeIOPSRead={
+              volumeIOPSRead
+                ? addMissingDataPoint(
+                    jointDataPointBaseonTimeSeries(volumeIOPSRead),
+                    queryStartingTime,
+                    7,
+                    1,
+                  )
+                : null
+            }
+            volumeIOPSWrite={
```

Comment:
I think that `volumeUsed `, `volumeStorageCapacity `, `volumeThroughputWrite `, `volumeThroughputRead `, `volumeIOPSRead ` and `volumeIOPSWrite ` should be defined inside `PerformanceGraphCard` and not in `VolumePage`.

It's a 1 to 1 relationship and add a lots of noise to the component for nothing. `PerformanceGraphCard` will become a `container` then.

If you want to do something reusable then you may want to do it a bit lower.

Reply:
Here you definitely right. I should make `PerformanceGraphCard` a container, and it's going to fetch its own data.